### ISSUE TITLE
any_errors_fatal still broken in ansible/stable-2.4

### ISSUE THREAD

**Issue by jeanfabrice** [ansible/ansible#30691]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
any_errors_fatal feature

##### ANSIBLE VERSION
```
ansible 2.4.0.0 (detached HEAD 6f461fbfd0) last updated 2017/09/21 14:10:44 (GMT +200)
```

##### CONFIGURATION

Default configuration

##### OS / ENVIRONMENT

macOS Sierra 10.12.6 (16G29)

##### SUMMARY
any_errors_fatal doesn't stop task execution. Basic usage

##### STEPS TO REPRODUCE
Create the following playbook:
```yaml
- hosts: all
  gather_facts: no
  any_errors_fatal: true

  tasks:
    - name: Fail a single host
      fail:
      when: inventory_hostname=="host0"
 
    - name: Run on both host
      debug: msg='Run on both host'
```

The accompanying hosts file is:
```
host0 ansible_host=127.0.0.1
host1 ansible_host=127.0.1.1
```
##### EXPECTED RESULTS

The playbook should stop at the first task for both host.

##### ACTUAL RESULTS

Playbook stops for host0, and continues to run the second task for host1 :

```
$ ansible-playbook playbook.yaml 

PLAY [all] ***************************************************************************************************************************************************************************************************************

TASK [Fail a single host] ************************************************************************************************************************************************************************************************
skipping: [host1]
fatal: [host0]: FAILED! => {"changed": false, "failed": true, "msg": "Failed as requested from task"}

TASK [Run on both host] **************************************************************************************************************************************************************************************************
ok: [host1] => {
    "msg": "Run on both host"
}

PLAY RECAP ***************************************************************************************************************************************************************************************************************
host0                      : ok=0    changed=0    unreachable=0    failed=1   
host1                      : ok=1    changed=0    unreachable=0    failed=0   
```

**Comment by mtnbikenc** [comment]
Just ran into this as well.

**Comment by wvidana** [comment]
Yes, upgraded hoping this would be fixed on 2.4, no luck.

**Comment by mtnbikenc** [comment]
I ran a git bisect on the stable-2.4 branch to track down where the error cropped up.  The bad commit is 1bf8c9997499361dc5fe05206969a4ff52a56ecf.

Here is the git bisect log:
```bash
> $ git bisect log
git bisect start
# bad: [ab4e83f288298fe05fdcf687b3ee8f65efa50702] Remove superfluous include on iosxr filesystems fact gathering (#21775)
git bisect bad ab4e83f288298fe05fdcf687b3ee8f65efa50702
# good: [9ab0a49ef2885a7ef216a92cf1bcabf3df1a305d] removes assert and uses to_list common function (#20866)
git bisect good 9ab0a49ef2885a7ef216a92cf1bcabf3df1a305d
# good: [b49d39c5dbdbf3faf959c3bdee10c245b502e1e3] Removes usage of expanduser because of type path
git bisect good b49d39c5dbdbf3faf959c3bdee10c245b502e1e3
# good: [a5d34f2ac2a88c821b5616d51de8d473c4715792] cloud: ovirt: add override_iptables to examples (#21474)
git bisect good a5d34f2ac2a88c821b5616d51de8d473c4715792
# bad: [ce9180b3a1c590e7f4404fbee3b85aeb479dcb13] roll up of vyos action fixes (#21642)
git bisect bad ce9180b3a1c590e7f4404fbee3b85aeb479dcb13
# bad: [e84713c76cdf052935c45134876e3d3471fb89d6] Revert "Remove the --nspawn-extra-args cli option"
git bisect bad e84713c76cdf052935c45134876e3d3471fb89d6
# good: [743d48af5cb842649a73b73daf82d8f0060a8d5e] Add Tower Project module (#21479)
git bisect good 743d48af5cb842649a73b73daf82d8f0060a8d5e
# bad: [1fc721118181be2ba5fd93ad61e10db9fa5eb166] Add a config section for systemd-nspawn driver
git bisect bad 1fc721118181be2ba5fd93ad61e10db9fa5eb166
# bad: [0ddcde2fb2e2e32763ac8cb5113ac17c0614b42e] updated docs to match
git bisect bad 0ddcde2fb2e2e32763ac8cb5113ac17c0614b42e
# bad: [1bf8c9997499361dc5fe05206969a4ff52a56ecf] Update any_errors_fatal field attributes in block/task to match play
git bisect bad 1bf8c9997499361dc5fe05206969a4ff52a56ecf
# good: [3b9892b45bab62d35547542f810371196f1dae82] New Modules for Ansible Tower (#21584)
git bisect good 3b9892b45bab62d35547542f810371196f1dae82
# first bad commit: [1bf8c9997499361dc5fe05206969a4ff52a56ecf] Update any_errors_fatal field attributes in block/task to match play
```

I dug into the problem further and found that it has something to do with setting the default for that attribute, `default=False` breaks it.  It works again after applying the following patch.  Note that the `_any_errors_fatal` attribute was moved to base.py.

```diff
diff --git a/lib/ansible/playbook/base.py b/lib/ansible/playbook/base.py
index b56151ad2..9b6bf6a06 100644
--- a/lib/ansible/playbook/base.py
+++ b/lib/ansible/playbook/base.py
@@ -165,7 +165,7 @@ class Base(with_metaclass(BaseMeta, object)):
     _ignore_errors       = FieldAttribute(isa='bool')
     _check_mode          = FieldAttribute(isa='bool')
     _diff                = FieldAttribute(isa='bool')
-    _any_errors_fatal     = FieldAttribute(isa='bool', default=C.ANY_ERRORS_FATAL, always_post_validate=True)
+    _any_errors_fatal    = FieldAttribute(isa='bool', always_post_validate=True)
 
     # param names which have been deprecated/removed
     DEPRECATED_ATTRIBUTES = [
                                                                                                                                              
```
This bug was fixed on the stable-2.3 branch with this commit, https://github.com/ansible/ansible/commit/8c2e9257d4b7fc248c40ba03945654e161b7cb1b.  Need to pull this over to devel and stable-2.4.
